crates/rpc/rpc-eth-api/src/helpers/pending_block.rs

```diff
@@ -122,14 +123,24 @@ pub trait LoadPendingBlock: EthApiTypes {
     /// Returns the locally built pending block
     fn local_pending_block(
         &self,
-    ) -> impl Future<Output = Result<Option<SealedBlockWithSenders>, Self::Error>> + Send
+    ) -> impl Future<Output = Result<Option<(SealedBlockWithSenders, Vec<Receipt>)>, Self::Error>> + Send
```

Comment:
let's not change the signature here. the pending block receipts field can be read inside scope of `load_block_and_receipts`. that way you also save the clone on the receipts which you are doing now to write it to the pending block.